knossos/repo.py

```diff
@@ -379,10 +389,10 @@ def process_pkg_selection(self, pkgs, recursive=True):
                         remains.append(v)
 
                 if len(remains) == 0:
-                    const = ','.join(['%s (%s)' % (s, m['owner'].title) for s, m in variants.items()])
-                    mod_titles = list(variants.values())[0]['mod'].title
-                    raise PackageConstraintNotMet('No version of mod "%s" found for these constraints: %s'
-                        % (mod_titles, const), mid, mod_titles, const)
+                    const = '\n'.join(['Package "%s" in %s (v%s) requires v%s of "%s" in %s' % (m['pack'],m['owner'].title,m['owner'].version,s,m['dep'].name, m['dep'].get_mod().title) for s, m in variants.items()])
```

Comment:
I think "requires v%s" should be "requires version %s" or something, because that field is actually a semver constraint term with its own syntax and you get weirdness like "v~4.6.0" as-is.